### PR TITLE
[release/9.0] Enable Cosmos vector search on EF9

### PR DESCRIPTION
**Pull request by ajcvickers**
This ports #35074 to 9.0-staging.

### Description

During the development of EF9, we worked with the Cosmos SDK folks to enable vector search against Cosmos DB using EF Core. This shipped in EF9, but we had to disable some functionality because it was not yet available in a non-preview Cosmos SDK. Cosmos have now released a public SDK, so this change updates EF9 to use these new APIs. The Cosmos folks have asked us to do this to make adoption of vector search easier for Cosmos customers.

### Customer impact

Customers who want to try out vector search can use EF and the Cosmos emulator as an easy, low-friction way to get into vector search on Cosmos.

### How found

N/A

### Regression

No.

### Testing

Existing tests enabled and passing.

### Risk

Low.

### CODE REVIEW

File: Directory.Packages.props [hunk 1]
```diff
@@ -36,7 +36,7 @@
     <PackageVersion Include="Microsoft.DotNet.Build.Tasks.Templating" Version="$(MicrosoftDotNetBuildTasksTemplatingVersion)" />
 
     <!-- Azure SDK for .NET dependencies -->
-    <PackageVersion Include="Microsoft.Azure.Cosmos" Version="3.43.0" />
+    <PackageVersion Include="Microsoft.Azure.Cosmos" Version="3.45.0" />
```
Comment (ajcvickers) [inline Directory.Packages.props:39]: @SamMonoRT @roji @Pilchie Should we go straight to 3.46 here, since it is out?

Comment (Pilchie) [reply Directory.Packages.props:39]: I'll leave that up to you, but note that 3.46.0 does include a build [breaking change](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4839), that you'd want to react to as well.

If you're going to stay on 3.45, consider using 3.45.2 though.

Comment (Pilchie) [reply Directory.Packages.props:39]: It might be nice to take the break as early as possible though.

**Review by Pilchie — APPROVED** [review]

File: Directory.Packages.props (hunk 1 above)
Comment (ajcvickers) [reply Directory.Packages.props:39]: Except we probably don't want to break in a patch.

Comment (Pilchie) [reply Directory.Packages.props:39]: True - I guess you're in a fine position if you reference 3.45.2 and if a customer wants to pin something higher, they get the break and can react there.

**Review by roji — APPROVED** [review]

**Comment by roji** [comment]
Servicing approved via email

**Pull request merged**